WordPress/Classes/Utility/Editor/GutenbergSettings.swift

```diff
@@ -68,12 +75,17 @@ class GutenbergSettings {
 
     /// True if gutenberg editor has been enabled at least once on the given blog
     func wasGutenbergEnabledOnce(for blog: Blog) -> Bool {
-        return blog.editor.mobile != nil
+        // If gutenberg was "globaly" enabled before, will take precedence over the new per-site flag
+        return database.object(forKey: Key.appWideEnabled) != nil || database.object(forKey: Key.enabledOnce(for: blog)) != nil
     }
 
     /// True if gutenberg should be autoenabled for the blog hosting the given post.
     func shouldAutoenableGutenberg(for post: AbstractPost) -> Bool {
-        return  post.containsGutenbergBlocks() && !wasGutenbergEnabledOnce(for: post.blog) && post.blog.editor.web == .gutenberg
+        return !wasGutenbergEnabledOnce(for: post.blog)
```

Comment:
I think there are some possible scenarios where gutenberg might be enabled but `wasGutenbergEnabledOnce` would be false (e.g. migration runs for a WordPress.com site with the app setting being `nil`). Should this bail early and return `false` if gutenberg is already enabled for this site?

Reply:
I think this is part of what it's explained here: https://github.com/wordpress-mobile/WordPress-iOS/pull/12189#issuecomment-515111020

In the scenario you mention, the user got gutenberg auto-enabled via the migration, so we do want to show the informative dialog in this case, even on a new post

Probably with the many changes on this behavior, the naming got confusing 🤔